Physicist/Physicist/Actor/Actor.cs

```diff
@@ -297,5 +342,79 @@ protected virtual bool OnCollision(Fixture fixtureA, Fixture fixtureB, FarseerPh
 
             return this.IsEnabled;
         }
+
+        protected void PlaySound(string name)
+        {
+            List<string> removelist = new List<string>();
+
+            // check for every Guid in the sound controller.
+            for (int i = 0; i < this.PlayingSounds.Count; i++)
+            {
+                if (!SoundController.ContainsInstance(this.PlayingSounds.ElementAt(i).Value))
+                {
+                    // add to a removelist
+                    removelist.Add(this.PlayingSounds.ElementAt(i).Key);
+                }
+            }
+
+            // remove everything in the removelist
+            foreach (string s in removelist)
+            {
+                this.PlayingSounds.Remove(s);
+            }
+
+            Guid newGuid;
+            newGuid = SoundController.PlayLocalizedSound(name, this.audioEmitter);
+
+            if (this.PlayingSounds.ContainsKey(name))
+            {
+                this.PlayingSounds[name] = newGuid;
+            }
+            else
+            {
+                this.PlayingSounds.Add(name, newGuid);
+            }
+        }
+
+        protected void PlaySound(string name, bool localized, bool looping, float volume, float pitch, float pan)
```

Comment:
You can replace the first version of PlaySound() with
PlaySound(string name)
{
    this.PlaySound(name, <defaults for localized, looping, etc>);
}

to reduce reuse